### PR TITLE
Use CentOS 6 package on Amazon Linux

### PR DESCRIPTION
**Pull request by michaelklishin**
## Proposed Changes

Also treat platform family 'amazon' the same as RHEL
since as of Chef 13 it's [separate from RHEL and Fedora](https://docs.chef.io/deprecations_ohai_amazon_linux.html).

## Types of Changes

- [x] Bug fix (non-breaking change which fixes issue #493)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)

## Checklist

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in related repositories

## Further Comments

#493.

### CODE REVIEW

**Comment by michaelklishin** [comment]
@neil-greenwood any chance you can give this PR a try in your environment? Thank you!

**Comment by neil-greenwood** [comment]
This doesn't work with chef-solo v12.1.2. It still tries to use the el7 RPM.

**Comment by michaelklishin** [comment]
@neil-greenwood what Amazon Linux version are you testing with? I could use some help with finding a Vagrant box to continue investigating.

File: recipes/default.rb [hunk 1]
```diff
@@ -41,16 +41,19 @@ class Chef::Resource
...
                                "rabbitmq-server-#{version}-1.el6.noarch.rpm"
                              end
-when 'suse'
-  default_rpm_package_name = "rabbitmq-server-#{version}-1.suse.noarch.rpm"
-end
+                           # see https://docs.chef.io/deprecations_ohai_amazon_linux.html
+                           when 'amazon'
+                             "rabbitmq-server-#{version}-1.el6.noarch.rpm"
+                           when 'suse'
+                             default_rpm_package_name = "rabbitmq-server-#{version}-1.suse.noarch.rpm"
```
Comment (devsibwarra) [inline recipes/default.rb]: variable assignment can be removed here

**Comment by neil-greenwood** [comment]
I'm using Amazon Linux, 2017.09.1

There don't seem to be any Vagrant boxes available for it, but there are docker images at https://hub.docker.com/_/amazonlinux/

There is a series of blog posts explaining how someone made a box from Amazon Linux - last post is at https://www.geekandi.com/2014/04/13/amazon-linux-vagrant-box-images/ with links to the others and some old boxes.

Finally: https://app.vagrantup.com/boxes/search?utf8=%E2%9C%93&sort=downloads&provider=&q=2016.09

**Comment by michaelklishin** [comment]
I will apply this to `v5.x` directly, with tests if I can find a suitable Docker image.

**Pull request closed without merging**